src/lib.rs

```diff
@@ -50,14 +49,22 @@ use thiserror::Error;
 #[path = "linux.rs"]
 mod platform;
 
+#[cfg(target_os = "windows")]
+#[path = "windows.rs"]
+mod platform;
+
 use platform::Brightness as Inner;
+use std::collections::HashMap;
 
 /// Interface to get and set brightness
 #[async_trait]
 pub trait Brightness {
     /// Returns the device name
     async fn device_name(&self) -> Result<String, Error>;
 
+    /// Returns platform specific device info
+    async fn device_info(&self) -> Result<HashMap<String, String>, Error>;
```

Comment:
I am not a fan of stringly-typed interfaces. Please remove this method (which is only implemented on Windows) and consider defining a `BrightnessExt` trait for Windows-specific extensions (in the spirit of `std::os::windows::fs::FileExt`).

Reply:
I was planning on implementing this for Linux as well once I have the time & figure out how to do it.
For example as a user it would be nicer to have a monitor description nicer than just `ddcci5` which is what I get for the device name on Linux.

Reply:
But I can understand wanting to do the interface differently - do you have any suggestions?

Reply:
In my opinion, the description can potentially be seen as a cross-platform functionality. `device_id`, `device_key` and `device_string` seem more Windows-specific AFAICT, and they somewhat overlap with `device_name` which should be considered a unique identifier for the device.

In summary, and if that's ok for your use case, I'd prefer adding the following method to the `Brightness` trait:

```rust
async fn description(&self) -> Result<String, Error>;
```

The linux implementation can return an empty string, at least for now.

Do you need access to the other properties your PR exposes (id, key, string)?

Reply:
sort of, I am using these properties to provide a filter which the users of my application (on all platforms) can filter different monitors by. 
I was also considering at some point using the device_key to query more information about the device (e.g. serial number and manufacturer).

Reply:
Let's merge it as is, and I will add a trait for Windows-specific extensions afterwards as described earlier. This will allow you to be able to query this extra information.

Reply:
actually now I am thinking about it - I would be quite happy with platform specific trait extensions. Especially if we could have one for Linux in future as well which gets whatever relevant properties there are?
I'm quite happy to do it in a couple of days time, or otherwise feel free to merge as you say and add after :)

Reply:
I added a trait for Windows-specific extensions. I removed `device_string` and `device_id` as they are identical to `device_description` and `device_name`, at least on my laptop. Please let me know if that's an issue for your other project.

Reply:
For me the `device_id` is not identical to the others:
```
device_description: "SC650_S24C650BW / S24C650MW (Digital)"
device_id: "\\?\DISPLAY#SAM09CF#5&397a0ebd&0&UID4356#{e6f07b5f-ee97-4a90-b076-33f57bf4eaa7}"
device_key: "\Registry\Machine\System\CurrentControlSet\Control\Class\{4d36e96e-e325-11ce-bfc1-08002be10318}\0003"
device_name: "\\.\DISPLAY2\Monitor0"
device_string: "SC650_S24C650BW / S24C650MW (Digital)"
```